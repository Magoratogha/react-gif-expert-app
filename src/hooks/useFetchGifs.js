import { useEffect, useState } from "react";
import getGifs from "../helpers/getGifs";

export default function useFetchGifs(category, resultsLimit, apiKey) {
    const [state, setState] = useState({
        data: [],
        isLoading: true
    });

    useEffect(() => {
        getGifs(category, resultsLimit, apiKey).then((gifs) => setState({
            data: gifs,
            isLoading: false
        }));
    }, [ apiKey, category, resultsLimit ]);

    return state;
};