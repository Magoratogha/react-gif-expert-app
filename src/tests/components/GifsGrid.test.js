import { queryAllByAttribute, queryByAttribute, render, screen } from "@testing-library/react";
import GifGrid from "../../components/GifsGrid";
import useFetchGifs from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe('Test in <GifsGrid />', () => {
    
    const category = 'Test'; 

    test('should render the component properly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            isLoading: true
        });
        const { container } = render(<GifGrid category={ category } />);
        expect(container).toMatchSnapshot();
    });
    
    test('should render the component properly when useFetch hook returns an images set', () => {
        const gifsList = [{
            id: 'testId',
            title: 'testTitle',
            url: 'https://test-url.com/test.jpg'
        }];
        useFetchGifs.mockReturnValue({
            data: gifsList,
            isLoading: false
        });
        const { container } = render(<GifGrid category={ category } />);
        expect(screen.queryByText('Loading...')).toBeNull();
        expect(queryAllByAttribute('class', container, 'gif-card', { exact: false }).length).toBe(gifsList.length);
    });
});