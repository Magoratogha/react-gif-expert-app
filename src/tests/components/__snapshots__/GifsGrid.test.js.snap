// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifsGrid /> should render the component properly 1`] = `
<div>
  <h3>
    Test
  </h3>
  <p
    class="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    class="cards-grid"
  />
</div>
`;
